vega_sim/api/data.py

```diff
@@ -946,9 +946,21 @@ def positions_by_market(
                 market_info = data_raw.market_info(
                     market_id=pos.market_id, data_client=data_client
                 )
+
+            settlement_asset_id = (
+                market_info.tradable_instrument.instrument.future.settlement_asset
+            )
+            if len(settlement_asset_id) < 1:
```

Comment:
This is an odd check, can we just do `if not settlement_asset_id` ?

Reply:
Fixed this + 2 other places with similar conditions